File: simple-stack-example-basic-java-view/src/main/java/com/zhuinden/navigationexampleview/screens/DashboardKey.java
package com.zhuinden.navigationexampleview.screens;

import com.google.auto.value.AutoValue;
import com.zhuinden.navigationexampleview.R;
import com.zhuinden.navigationexampleview.core.navigation.BaseKey;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class DashboardKey extends BaseKey {
    public static DashboardKey create() {
        return new AutoValue_DashboardKey();
    }

    @Override
    public int layout() {
        return R.layout.dashboard_view;
    }
}

File: samples/basic-samples/simple-stack-example-basic-kotlin-view/src/main/java/com/zhuinden/navigationexamplekotlinview/screens/DashboardKey.kt
package com.zhuinden.navigationexamplekotlinview.screens

import com.zhuinden.navigationexamplekotlinview.R
import com.zhuinden.navigationexamplekotlinview.core.navigation.BaseKey
import kotlinx.parcelize.Parcelize

/**
 * Created by Owner on 2017. 06. 29..
 */
@Parcelize
data object DashboardKey : BaseKey() {
    override fun layout(): Int = R.layout.dashboard_view
}
